tests/examples/test_examples.py

```diff
@@ -38,6 +38,15 @@ def clean_up_mlflow_virtual_environments():
             shutil.rmtree(path)
 
 
+@pytest.fixture(scope="module", autouse=True)
+def mock_openai():
+    # Some examples includes OpenAI API calls, so we start a mock server.
+    with start_mock_openai_server() as base_url:
+        os.environ["OPENAI_API_BASE"] = base_url
+        os.environ["OPENAI_API_KEY"] = "test"
+        yield
```

Comment:
```suggestion
        with pytest.MonkeyPatch.context() as mp:
            mp.setenv("OPENAI_API_BASE", base_url)
            mp.setenv("OPENAI_API_KEY", "test")
            yield
```

to minimize side effects